bentoml/frameworks/pytorch.py

```diff
@@ -114,3 +130,103 @@ def get(self):
+
+    Args:
+        name (string): Name of the pytorch model
+    Raises:
+        MissingDependencyException: torch and pytorch_lightning package is required.
+
+    Example usage:
+
+    >>>
+    >>> # Train pytorch lightning model
+    >>> from pytorch_lightning.core.lightning import LightningModule
+    >>>
+    >>> class SimpleModel(LightningModule):
+    >>>     def forward(self, x):
+    >>>         return x.add(1)
+    >>>
+    >>> model = SimpleModel()
+    >>>
+    >>> import bentoml
+    >>> import torch
+    >>> from bentoml.adapters import JsonInput
+    >>> from bentoml.frameworks.pytorch import PytorchLightningModelArtifact
+    >>>
+    >>> @bentoml.env(infer_pip_packages=True)
+    >>> @bentoml.artifacts([PytorchLightningModelArtifact('model')])
+    >>> class PytorchLightingService(bentoml.BentoService):
+    >>>     @bentoml.api(input=DataframeInput(), batch=True)
+    >>>     def predict(self, df):
+    >>>         input_tensor = torch.from_numpy(df.to_numpy())
+    >>>         return self.artifacts.model(input).numpy()
+    >>>
+    >>> svc = PytorchLightingService()
+    >>> svc.pack('model', model)
+    >>>
+    >>> svc.save()
+    """
+
+    def __init__(self, name):
+        super(PytorchLightningModelArtifact, self).__init__(name)
+        self._model = None
+        self._model_path = None
+
+    def _saved_model_file_path(self, base_path):
+        return os.path.join(base_path, self.name + '.pt')
+
+    def pack(self, path_or_model, metadata=None):  # pylint:disable=arguments-differ
+        if _is_pytorch_lightning_model_file_like(path_or_model):
+            self._model_path = path_or_model
+        else:
+            try:
+                from pytorch_lightning.core.lightning import LightningModule
+            except ImportError:
+                raise InvalidArgument(
+                    '"pytorch_lightning.lightning.LightningModule" model is required '
+                    'to pack a PytorchLightningModelArtifact'
+                )
+            if isinstance(path_or_model, LightningModule):
+                self._model = path_or_model
+            else:
+                raise InvalidArgument(
+                    'a LightningModule model is required to pack a '
+                    'PytorchLightningModelArtifact'
+                )
+        return self
+
+    def load(self, path):
+        return self.pack(self._saved_model_file_path(path))
+
+    def set_dependencies(self, env: BentoServiceEnv):
+        env.add_pip_packages(['pytorch'])
+
+    def get(self):
+        if self._model is not None:
+            return self._model
+        else:
+            return self._get_torch_script_model()
+
+    def save(self, dst):
+        try:
+            import torch
+        except ImportError:
+            raise MissingDependencyException(
+                '"torch" package is required for saving Pytorch lightning model'
+            )
+        script = self._model.to_torchscript()
```

Comment:
when `pack` with a path, `self._model` is not defined, it should be copying over files


Make sure to cover all the cases in the test